lib/erlef/http.ex

```diff
@@ -0,0 +1,73 @@
+defmodule Erlef.HTTP do
+  @moduledoc """
+  Erlef.HTTP - Lightweight wrapper around hackney with a few json encoding/decoding conveniences
+  """
+
+  @success_range 200..299
+  @json_mimes ["application/json", "application/json; charset=utf-8", "application/vnd.api+json"]
+
+  def perform(method, url, headers, body, opts) do
+    {:ok, payload} = maybe_json_encode(content_type(headers), body)
+
+    case :hackney.request(method, url, headers, payload, build_options(opts)) do
```

Comment:
If any values in headers are ever user controlled, this could be an issue, because Hackney isn't header-injection safe. E.g. a header like {"token", "value\r\nother_token: other_value"} would parse as multiple headers. This could have different effects in different contexts.

Reply:
I can't imagine a situation where we would ever allow user controlled headers 🤔 If you mean headers that are driven via user input.

Reply:
For example, if you are using a user provided token as an auth header value or bearer token or something.

Reply:
I see. It would be responsibility of another module I think to ensure that doesn't happen. Fortunately we don't have a case like that right now... but it's interesting topic since http allows for duplicate headers. 
  
I think in this case it would be the responsibility of the Erlfef.Wildapricot module. Excellent call out!

Reply:
@GriffinMB ahh, I totally was in another space time. After coming back to this, it seems Erlef.HTTP is the appropriate place to guard against this 👍

Reply:
> If any values in headers are ever user controlled, this could be an issue, because Hackney > isn't header-injection safe. E.g. a header like {"token", "value\r\nother_token: other_value"} would parse as multiple headers. This could have different effects in different contexts.

would love to see a ticket fo it :) Patch should be easy.

Reply:
yeah, I'm not terribly sure it's a good patch for a generic http client. It depends, I could not find clarification on whether CRLF is illegal for header keys and/or values. None the less, we'll guard it again here by simple uri encoding the keys and values.

Reply:
Implemented. We simply enumerate the headers, uri encode the header key, split the value on " ", URI encode each part, re-join the value parts.